scripts/extract-synapse-values.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 New Vector Ltd
+#
+# SPDX-License-Identifier: AGPL-3.0-only
+
+set -xe
+
+
+targetDirectory="$1"
+homeserverconfig="$2"
+
+
+if [ -z "$targetDirectory" ]; then
+  echo "Please provide a valid path for the ESS-Helm values directory"
+  echo "Usage: $0 <target_directory> <homeserver.yaml>"
+  echo "Example: $0 ~/ess-helm-values /matrx/synapse/config/homeserver.yaml"
+  exit 1
+fi
+
+if [ -z "$homeserverconfig" ]; then
+  if [ -f "homeserver.yaml" ]; then
+    homeserverconfig="homeserver.yaml"
+  else
+    echo "Please provide a valid path for the homeserver.yaml file"
+    echo "try find / -name homeserver.yaml or copy the yaml file to the current directory"
+    echo "and run the script again"
+    echo "Usage: $0 <target_directory> <homeserver.yaml>"
+    echo "Example: $0 ~/ess-helm-values /matrx/synapse/config/homeserver.yaml"
+    exit 1
+  fi
+fi
+
+read -p "Is your synapse running in a docker container? (y/n): " isDocker
+if [[ "$isDocker" == "y" ]]; then
+  # Get the list of running containers
+  containers=$(docker ps --format '{{.Names}}')
+  echo "Select the Synapse container from the list:"
+  select synapseContainer in $containers; do
+    if [[ -n "$synapseContainer" ]]; then
+      echo "You selected: $synapseContainer"
+      break
+    else
+      echo "Invalid selection. Please try again."
+    fi
+  done
+fi
+
+
+if [ ! -d "$targetDirectory" ]; then
+  mkdir -p "$targetDirectory"
+fi
+
+touch "$targetDirectory/hostnames.yaml"
+touch "$targetDirectory/secrets.yaml"
+
+servername=$(yq '.servername' "$homeserverconfig")
+synapseHost=$(yq '(.public_baseurl | match "https://(.+)/").captures[0].string' "$homeserverconfig")
+macaroon=$(yq '.macaroon_secret_key' "$homeserverconfig")
```

Comment:
Synapse supports  `_path` suffixes for all secrets nowadays https://element-hq.github.io/synapse/latest/usage/configuration/config_documentation.html?highlight=macaroon#macaroon_secret_key_path . It would be great if the script could handle those automatically.